compiler/rustc_passes/src/stability.rs

```diff
@@ -164,8 +167,32 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
 
         let (stab, const_stab) = attr::find_stability(&self.tcx.sess, attrs, item_sp);
 
-        let const_stab = const_stab.map(|(const_stab, _)| {
+        let const_stab = const_stab.map(|(const_stab, const_span)| {
             let const_stab = self.tcx.intern_const_stability(const_stab);
+            if let Some(fn_sig) = fn_sig {
+                if !(fn_sig.header.constness == hir::Constness::Const
+                    && fn_sig.header.abi != Abi::RustIntrinsic
+                    && fn_sig.header.abi != Abi::PlatformIntrinsic)
+                {
+                    self.tcx
+                        .sess
+                        .struct_span_err(
+                            fn_sig.span,
+                            "attributes `#[rustc_const_unstable]` \
+                            and `#[rustc_const_stable]` require \
+                            the function or method to be marked `const`",
+                        )
+                        .span_suggestion_short(
+                            fn_sig.span,
+                            "make the function or method const",
+                            String::new(),
+                            rustc_errors::Applicability::Unspecified,
+                        )
```

Comment:
If you don't provide a suggestion, you should use `span_help` instead.
```suggestion
                        .span_help(fn_sig.span, "make the function or method const")
```